### DISCUSSION TITLE
Token.actions.githubusercontent.com returns incorrect token

### DISCUSSION THREAD

**Discussion by TROEERI** [community/community#25223] [Actions]
<p>We are using GitHub Actions AWS federation. After an update on Oct. 6th 2021 the URL to request the OIDC token has been changed to <code>token.actions.githubusercontent.com</code>.</p>
<p>The token returned still has iss/aud  <code>vstoken.actions.githubusercontent.com</code>, which was the request URL before the change (see bottom of this post).</p>
<p>This now breaks our ODIC connection to AWS:<br>
“An error occurred (InvalidIdentityToken) when calling the AssumeRoleWithWebIdentity operation: Incorrect token audience”</p>
<p>Does anybody know how to deal with this issue?</p>
<pre><code class="lang-auto">{
  "nameid": "dddddddd-dddd-dddd-dddd-dddddddddddd",
  "scp": "Actions.GenericRead:00000000-0000-0000-0000-000000000000 Actions.UploadArtifacts:00000000-0000-0000-0000-000000000000/1:Build/Build/1205 DistributedTask.GenerateIdToken:74494030-3856-426f-9c1c-2ed8806a371a:6e2e6d2e-3d07-5d27-f52a-c3c85fbb7c29 LocationService.Connect ReadAndUpdateBuildByUri:00000000-0000-0000-0000-000000000000/1:Build/Build/1205",
  "IdentityTypeClaim": "System:ServiceIdentity",
  "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/sid": "DDDDDDDD-DDDD-DDDD-DDDD-DDDDDDDDDDDD",
  "http://schemas.microsoft.com/ws/2008/06/identity/claims/primarysid": "dddddddd-dddd-dddd-dddd-dddddddddddd",
  "aui": "xyz",
  "sid": "xyz",
  "ac": "[{\"Scope\":\"refs/heads/main\",\"Permission\":3}]",
  "oidc_sub": "repo:fielmann-ag/kls:ref:refs/heads/main",
  "oidc_extra": "{\"ref\":\"refs/heads/main\",\"sha\":\"6abb1f42030e5d4fdc6b2e2d15419cd097add694\",\"repository\":\"fielmann-ag/kls\",\"repository_owner\":\"fielmann-ag\",\"run_id\":\"1316676112\",\"run_number\":\"22\",\"run_attempt\":\"1\",\"actor\":\"TROEERI\",\"workflow\":\"Deploy grafana dashboards\",\"head_ref\":\"\",\"base_ref\":\"\",\"event_name\":\"push\",\"ref_type\":\"branch\",\"job_workflow_ref\":\"fielmann-ag/kls/.github/workflows/eks-grafana-dashboards.yml@refs/heads/main\"}",
  "orchid": "74494030-3856-426f-9c1c-2ed8806a371a.deploy-to-dev.__default",
  "iss": "vstoken.actions.githubusercontent.com",
  "aud": "vstoken.actions.githubusercontent.com|vso:xyz",
  "nbf": 1633617542,
  "exp": 1633619342,
  "alg": "HS256"
}
</code></pre>

**Comment by Frisch12** [comment]
<p>Have the same issue!<br>
Would be nice to get a fix for it as the Tokens that are generated by “<a href="http://token.actions.githubusercontent.com" rel="noopener nofollow ugc">token.actions.githubusercontent.com</a>” are generally invalid!</p>

**Comment by tomislacker** [comment]
<p><a class="mention" href="/troeeri">@TROEERI</a> How were you able to find that JSON blob? I’m facing a similar situation and have a vested interest in troubleshooting further.</p>

**Comment by TROEERI** [comment, marked as answer]
<p>The issue has been fixed by GitHub. Please have a look at the official documentation:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/72204573/181089679-7427ddcf-30cd-4556-8af1-5fc9ddf761b0.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services" target="_blank" rel="noopener nofollow ugc">Configuring OpenID Connect in Amazon Web Services - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
